src/client/src/orangehrmCorporateDirectoryPlugin/pages/CorporateDirectory.vue

```diff
@@ -0,0 +1,253 @@
+      employeeJobTitle: item.jobTitle?.title,
+      employeeSubUnit: item.subunit?.name,
+      employeeLocation: item.location?.name,
+    };
+  });
+};
+export default {
+  name: 'CorporateDirectory',
+
+  components: {
+    'employee-autocomplete': EmployeeAutocomplete,
+    'jobtitle-dropdown': JobtitleDropdown,
+    'subunit-dropdown': SubunitDropdown,
+    'summary-card': SummaryCard,
+    'summary-card-details': SummaryCardDetails,
+    'employee-details': EmployeeDetails,
+    'oxd-loading-spinner': Spinner,
+  },
+
+  setup() {
+    const http = new APIService(
+      'https://07bd2c2f-bd2b-4a9f-97c7-cb744a96e0f8.mock.pstmn.io',
+      'api/v2/corporate-directory/employees',
+    );
+    const limit = 10; // this is a static limit since no pagination
+    const state = reactive({
+      employees: [],
+      total: 0,
+      colSize: 4,
+      windowWidth: 0,
+      currentIndex: -1,
+      isLoading: false,
+      offset: 0,
+    });
+
+    const fetchData = () => {
+      state.isLoading = true;
+      http
+        .getAll({
+          limit: limit,
+          offset: state.offset,
+        })
+        .then(response => {
+          const {data, meta} = response.data;
+          state.total = meta?.total || 0;
+          if (Array.isArray(data)) {
+            state.employees = [
+              ...state.employees,
+              ...employeeDataNormalizer(data),
+            ];
+          }
+        })
+        .finally(() => (state.isLoading = false));
+    };
+
+    const {scrollerRef} = useInfiniteScroll(() => {
+      if (state.employees.length >= state.total) return;
+      state.offset += limit;
+      fetchData();
+    });
+
+    return {
+      scrollerRef,
+      ...toRefs(state),
+      fetchData,
+    };
+  },
+
+  computed: {
+    isMobile() {
+      return this.windowWidth < 600 ? true : false;
+    },
+    isTotalZero() {
+      return this.total <= 0 ? true : false;
+    },
+  },
+
+  beforeMount() {
+    this.fetchData();
+
+    this.onResize();
+    window.addEventListener('resize', this.onResize);
+  },
+  beforeUnmount() {
+    window.removeEventListener('resize', this.onResize);
+  },
+  methods: {
+    hideEmployeeDetails() {
+      this.currentIndex = -1;
+      this.colSize = 4;
+    },
+    showEmployeeDetails(index) {
+      if (this.currentIndex != index) {
+        this.currentIndex = index;
+        this.colSize = 3;
+      } else {
+        this.hideEmployeeDetails();
+      }
+    },
+    onResize() {
+      this.windowWidth = window.innerWidth;
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.orangehrm-corporate-directory {
+  display: flex;
+
+  &-sidebar {
+    margin-left: 16px;
+  }
+}
+
+@import '@ohrm/oxd/styles/_mixins.scss';
+.orangehrm-container {
+  overflow: auto;
+  max-height: 512px;
+  @include oxd-scrollbar();
+  &-loader {
+    margin: 0 auto;
+    background-color: $oxd-white-color;
+  }
+}
+
+@media (max-width: 600px) {
```

Comment:
@Super-Chama can't we use respond-to('md')

Reply:
yes, mentioned it

Reply:
done